lib/chef-dk/policyfile/remote_lock_fetcher.rb

```diff
@@ -41,6 +60,32 @@ def errors
         error_messages
       end
 
+      # @return [Hash] The source_options that describe how to fetch this exact lock again
+      def source_options_for_lock
+        source_options
+      end
+
+      # Applies source options from a lock file. This is used to make sure that the same
+      # policyfile lock is loaded that was locked
+      #
+      # @param options_from_lock [Hash] The source options loaded from a policyfile lock
+      def apply_locked_source_options(options_from_lock)
+        # There are no options the lock could provide
+      end
+
+      # @return [String] of the policyfile lock data
+      def lock_data
+        fetch_lock_data.tap do |data|
+          validate_revision_id(data["revision_id"], source_options)
+          data["cookbook_locks"].each do |cookbook_name, cookbook_lock|
+            cookbook_path = cookbook_lock["source_options"]["path"]
```

Comment:
I think this seems reasonable to ensure people aren't trying to mix local and remote lock files